### PR TITLE
[storekit] Add nullability to (generated and manual) bindings

### PR DESCRIPTION
**Pull request by tj-devel709**
This PR aims to bring nullability changes to StoreKit.
Following the steps [here](https://github.com/xamarin/xamarin-macios/wiki/Nullability):

1. I am adding `nullable enable` to all manual files that are not "API_SOURCES" in src/frameworks.sources and making the required nullability changes
2. Changing any `== null` or `!= null` to `is null` and `is not null`

### CODE REVIEW

File: src/StoreKit/SKCloudServiceSetupOptions.cs [hunk 1]
```diff
@@ -10,10 +12,12 @@ partial class SKCloudServiceSetupOptions {
 
 		public virtual SKCloudServiceSetupAction? Action {
 			get {
-				return (SKCloudServiceSetupAction?) (SKCloudServiceSetupActionExtensions.GetValue (_Action));
+				if (_Action is null)
+					ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (_Action));
+				return (SKCloudServiceSetupAction?) (SKCloudServiceSetupActionExtensions.GetValue (_Action!));
```
Comment (tj-devel709) [inline src/StoreKit/SKCloudServiceSetupOptions.cs:17]: I know that we should not throw an exception in a getter, but the `SKCloudServiceSetupActionExtensions.GetValue` throws an exception if the _Action is null anyways... Suggestion?

File: src/StoreKit/SKCloudServiceSetupOptions.cs [hunk 2]
```diff
@@ -10,10 +12,12 @@ partial class SKCloudServiceSetupOptions {
 
 		public virtual SKCloudServiceSetupAction? Action {
 			get {
-				return (SKCloudServiceSetupAction?) (SKCloudServiceSetupActionExtensions.GetValue (_Action));
+				if (_Action is null)
+					ObjCRuntime.ThrowHelper.ThrowArgumentNullException (nameof (_Action));
```
Comment (mandel-macaque) [inline src/StoreKit/SKCloudServiceSetupOptions.cs:16]: SKCloudServiceSetupAction? allows to return a null, just return null if _Action is null piece of 🍰 :)

**Review by mandel-macaque — CHANGES_REQUESTED** [review]

File: src/StoreKit/SKCloudServiceSetupOptions.cs (hunk 2 above)
Comment (rolfbjarne) [reply src/StoreKit/SKCloudServiceSetupOptions.cs:16]: Agreed, just return `null` here instead, and no exception will need to be thrown.

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]
After @mandel-macaque 👍

**Comment by tj-devel709** [comment]
@mandel-macaque @rolfbjarne Updated!

**Review by rolfbjarne — APPROVED** [review]

**Comment by tj-devel709** [comment]
This one is ready to merge after approval from @mandel-macaque

**Pull request merged**